scipy/spatial/tests/test_kdtree.py

```diff
@@ -217,46 +217,19 @@ def test_vectorized_query_multiple_neighbors(self):
         assert_(np.all(i[:, :, -s:] == self.kdtree.n))
 
     @pytest.mark.parametrize('r', [0.8, 1.1])
-    def test_single_query_all_neighbors(self, r):
+    def test_single_query_raises_for_k_none(self, r):
```

Comment:
There is no need to have 2 tests and the parametrization. The check for k=None is at the beginning of the function so other parameters don't have any impact.

Reply:
Right. I removed both the parametrization and the second test.